fluentlenium-core/src/main/java/org/fluentlenium/adapter/FluentAdapter.java

```diff
@@ -91,60 +101,33 @@ public void releaseFluent() {
     }
 
     /**
-     * Override this method to change the driver
-     *
-     * @return returns WebDriver which is set to FirefoxDriver by default - can be overwritten
+     * @return A new WebDriver instance.
+     * @see #getDriver()
+     * @deprecated Override {@link #newWebDriver()} instead, or consider using {@link ConfigurationProperties#getWebDriver()}.
      */
+    @Deprecated
     public WebDriver getDefaultDriver() {
-        return new FirefoxDriver();
+        WebDriver webDriver = WebDrivers.INSTANCE.newWebDriver(getWebDriver(), getCapabilities());
+        if (Boolean.TRUE.equals(getEventsEnabled())) {
+            webDriver = new EventFiringWebDriver(webDriver);
+        }
+        return webDriver;
     }
 
     /**
-     * Override this method to set the base URL to use when using relative URLs
+     * Creates a new {@link WebDriver} instance.
+     *
+     * This method should not be called by end user, but may be overriden if required.
+     *
+     * Before overriding this method, you should consider using {@link WebDrivers} registry and configuration
+     * {@link ConfigurationProperties#getWebDriver()}.
      *
-     * @return The base URL, or null if relative URLs should be passed to the driver untouched
+     * To retrieve the current managed {@link WebDriver}, call {@link #getDriver()} instead.
+     *
+     * @return A new WebDriver instance.
+     * @see #getDriver()
      */
-    public String getDefaultBaseUrl() {
-        return null;
-    }
-
-    @Override
-    public void setScreenshotPath(String path) {
-        this.screenshotPath = path;
-    }
-
-    @Override
-    public void setHtmlDumpPath(String htmlDumpPath) {
-        this.htmlDumpPath = htmlDumpPath;
-    }
-
-    @Override
-    public void setScreenshotMode(TriggerMode mode) {
-        this.screenshotMode = mode;
-    }
-
-    @Override
-    public TriggerMode getScreenshotMode() {
-        return screenshotMode;
-    }
-
-    @Override
-    public String getScreenshotPath() {
-        return screenshotPath;
-    }
-
-    @Override
-    public String getHtmlDumpPath() {
-        return htmlDumpPath;
-    }
-
-    @Override
-    public void setHtmlDumpMode(TriggerMode htmlDumpMode) {
-        this.htmlDumpMode = htmlDumpMode;
-    }
-
-    @Override
-    public TriggerMode getHtmlDumpMode() {
-        return htmlDumpMode;
+    public WebDriver newWebDriver() {
```

Comment:
getNewWebDriver/setNewWebDriver would be for Java Bean property, but this is not a Java Bean property. It's a factory method to create new WebDriver instance, so that's why I gave this name instead of getDefaultDriver() which was confusing to some people.